Doc/tutorial/introduction.rst

```diff
@@ -435,6 +428,30 @@ the :meth:`!list.append` *method* (we will see more about methods later)::
    >>> cubes
    [1, 8, 27, 64, 125, 216, 343]
 
+When you assign a variable to a list or any other object,
```

Comment:
I'd like to start with a strong general statement:

"Assignment in Python never copies data. When you assign a list to a variable, the variable refers to the existing list. Any changes you make through one variable will be seen through the other variables that refer to it::"

Reply:
@nedbat *"Assignment in Python never copies data"* - What about `a[:] = b`?

Reply:
> What about `a[:] = b` or `[*a] = b`?

Those copy references, not data.  It's still multiple references to the same objects:
```
>>> a = [1, 2, 3]
>>> b = [{1}, {2}, {3}]
>>> a[:] = b
>>> a
[{1}, {2}, {3}]
>>> b
[{1}, {2}, {3}]
>>> b[0].add(99)
>>> a
[{1, 99}, {2}, {3}]
>>> b
[{1, 99}, {2}, {3}]
```

But to keep this example focused, how about "Simple assignment never copies data"?

Reply:
*"Those copy references, not data"* - With that view, how do you *ever* copy data? Does `list(b)` likewise not copy data? What even *is* data then? Anyway, `mybytearray[:] = mybytes` then. But yes, with "Simple" it sounds good to me.